hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordPayload.java

```diff
@@ -116,7 +121,15 @@ default Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord currentValu
    */
   @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
   default Option<IndexedRecord> getInsertValue(Schema schema, Properties properties) throws IOException {
-    return getInsertValue(schema);
+    boolean dropPartCol = Boolean.parseBoolean(properties.getProperty(HoodieTableConfig.DROP_PARTITION_COLUMNS.key()));
```

Comment:
this won't be effective on some default payload classes like `DefaultHoodieRecordPayload`. can we audit the subclasses and fix altogether?

Reply:
have fix all subclasses which extends `HoodieRecordPayload` and overwrite `getInsertValue(Schema schema, Properties properties) `，Except `ExpressionPayload.java`, this payload is used for spark sql, and it works fine with `hoodie.datasource.write.drop.partition.columns =true, ` 
in my opinion, there's no need to change it. what do you think?